internet/definitions.go

```diff
@@ -33,12 +37,83 @@ type StackNode interface {
 
 // node is a concrete StackNode as stored in Stacks. Methods are devirtualized for performance benefits, especially on TinyGo.
 type node struct {
-	currConnID  uint64
-	connID      *uint64
-	demux       func([]byte, int) error
-	encapsulate func([]byte, int) (int, error)
-	proto       uint16
-	port        uint16
+	currConnID       uint64
+	connID           *uint64
+	demux            func([]byte, int) error
+	checkEncapsulate func(*internal.EncData) bool
+	doEncapsulate    func([]byte, int) (int, error)
+	proto            uint16
+	port             uint16
+}
+
+type handlers struct {
+	nodes   []node
+	current uint
+}
+
+func (h *handlers) Len() int {
+	return len(h.nodes)
+}
+
+func (h *handlers) GetCurrent() (current *node) {
+	if h.current < uint(len(h.nodes)) {
+		return &h.nodes[h.current]
+	}
+	return nil
+}
+
+func (h *handlers) GetNext() (next *node) {
```

Comment:
We can call these methods `Next` and `Current` and call the variable `handlers` in code so it reads nicely: handlers.Current, handlers.Next